apm/src/app/user/login/login.component.ts

```diff
@@ -0,0 +1,24 @@
+import { Component } from '@angular/core';
+import { Router } from '@angular/router';
+import { UserService } from '../user.service';
+
+@Component({
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.css']
+})
+
+export class LoginComponent{
+  userName: any;
+  password: any; 
+  homepageUrl: string = "/welcome";
```

Comment:
just a suggestion, typescript is smart enough to conclude the type when a value is assigned, so we can ignore using the type here.